### PR TITLE
Integrations moderation

### PR DESCRIPTION
**Pull request by mario**
This adds moderation options to our new integrations submission flow.

As a superuser inside the Account interface you'll see the usual 'Integrations' page where, if there are any pending integrations, you'll have a table view of those with the ability to view them and afterwards, approve or reject.

It'll also send email to the marketing team upon submission. Please note that atm you need superuser privileges to approve an integration - so please give members (or one/two at least) those permissions.

As always, the usual warnings apply: code has been tested and works for me. It's not super clean or great, but it gets the job done at this moment with a decent UI.

### CODE REVIEW

**Comment by mario** [comment]
Fixes https://github.com/nextcloud/appstore/issues/681

**Comment by mario** [comment]
Screenshot 2020-04-18 at 14 00 11

**Comment by jospoortvliet** [comment]
epic stuff, Mario. I can't judge the code but it is what we need from a feature prespective! :+1:

File: nextcloudappstore/scaffolding/forms.py [hunk 1]
```diff
@@ -58,8 +59,6 @@ class AppScaffoldingForm(Form):
 class IntegrationScaffoldingForm(Form):
     name = CharField(max_length=80, label=_('Integration name'),
                      widget=TextInput())
-    author_name = CharField(max_length=80, label=_('Author\'s full name'))
-    author_email = EmailField(label=_('Author\'s e-mail'))
```
Comment (blizzz) [inline nextcloudappstore/scaffolding/forms.py:62]: remove intentionally?

File: nextcloudappstore/scaffolding/forms.py [hunk 2]
```diff
@@ -86,29 +85,39 @@ class IntegrationScaffoldingForm(Form):
...
+            if app.can_update(user) or user.is_superuser:
+                if action == "reject":
+                    Screenshot.objects.filter(app=app).delete()
+                    app.delete()
+                elif action == "approve":
+                    app.approved = True
+                    if settings.DISCOURSE_TOKEN:
+                        self._create_discourse_category(app_id)
+                    app.save()
+                    return app_id
```
Comment (blizzz) [inline nextcloudappstore/scaffolding/forms.py:102]: does it mean you as intgeration-developer can you self-approve?

File: nextcloudappstore/scaffolding/views.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
-from django.http import HttpResponse
-from django.views.generic import FormView
+from django.http import HttpResponse, Http404
```
Comment (blizzz) [inline nextcloudappstore/scaffolding/views.py:1]: didn't you use `HttpNotFound` or so a few files above?

File: nextcloudappstore/settings/base.py [hunk 4]
```diff
@@ -310,3 +310,4 @@
 
 # Nextcloud Email
 NEXTCLOUD_FROM_EMAIL = 'appstore@nextcloud.com'
+NEXTCLOUD_INTEGRATIONS_APPROVAL_EMAILS = ['marketing-team@nextcloud.com']
```
Comment (blizzz) [inline nextcloudappstore/settings/base.py:313]: do they know about their luck? :D

Comment (mario) [reply nextcloudappstore/settings/base.py:313]: Yes :P

File: nextcloudappstore/scaffolding/forms.py (hunk 1 above)
Comment (mario) [reply nextcloudappstore/scaffolding/forms.py:62]: Yes. It wasn't used - the full name and email are re-used from the account that submits it.

File: nextcloudappstore/scaffolding/forms.py (hunk 2 above)
Comment (mario) [reply nextcloudappstore/scaffolding/forms.py:102]: Uh, good catch - will check.

File: nextcloudappstore/scaffolding/views.py (hunk 3 above)
Comment (mario) [reply nextcloudappstore/scaffolding/views.py:1]: I did add it it, but it wasn't used. I just need to remove the import there - will do.

**Comment by mario** [comment]
@blizzz updated.

File: nextcloudappstore/scaffolding/views.py (hunk 3 above)
Comment (blizzz) [reply nextcloudappstore/scaffolding/views.py:1]: I was mostly confused by different naming for seemingly the same thing.

**Comment by blizzz** [comment]
it's deployed :sparkles:

**Comment by jospoortvliet** [comment]
> it's deployed

rocking!

**Pull request merged**